nifi-nar-bundles/nifi-oraclecdc-bundle/nifi-oraclecdc-processors/src/main/java/org/apache/nifi/processors/oraclecdc/OracleChangeCapture.java

```diff
@@ -0,0 +1,251 @@
+            .build();
+    public static final Relationship UNMATCHED = new Relationship.Builder().name("UNMATCHED")
+            .description("UNSUPPORTED OPERATION").build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    protected DBCPService dbcpService;
+
+    protected Object xsOut;
+
+    private OracleCDCService cdcService;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(XS_OUT);
+        descriptors.add(CDC_SERVICE);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(INSERTS);
+        relationships.add(UPDATES);
+        relationships.add(DELETES);
+        relationships.add(UNMATCHED);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        byte[] position = null;
+        final StateManager stateManager = context.getStateManager();
+        StateMap stateMap;
+        try {
+            stateMap = stateManager.getState(Scope.CLUSTER);
+        } catch (final IOException ioe) {
+            stateMap = null;
+            getLogger().warn("Failed to retrieve the low watermark from State Manager", ioe);
+        }
+        if (stateMap != null) {
+            System.out.println("position from statemap" + stateMap.get("positon"));
+            position = new Base32(true).decode(stateMap.get("positon"));
+        }
+        cdcService = context.getProperty(CDC_SERVICE).asControllerService(OracleCDCService.class);
+        String xsOutServerName = context.getProperty(XS_OUT).getValue();
+        xsOut = cdcService.attach(xsOutServerName, position);
+
+    }
+
+    @OnShutdown
+    @OnStopped
+    public void shutdown(final ProcessContext context) {
+        cdcService.detach(xsOut);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        CDCHandler handler = new CDCHandler(session);
+        cdcService.receiveEvents(xsOut, handler);
+        if (handler.isMoveMarker()) {
+            cdcService.setProcessedLowWaterMark(xsOut, handler.getLastPosition());
+
+            final StateManager stateManager = context.getStateManager();
+            try {
+                HashMap<String, String> stateMap = new HashMap<String, String>();
+                System.out.println(
```

Comment:
Do we still need this `System.out.println`?

Reply:
removed it